src/App.jsx

```diff
@@ -1,7 +1,32 @@
+import { useState } from "react";
 import "./App.css";
+import Gnb from "./components/headers/Gnb.jsx";
+import RestaurantCategoryFilter from "./components/mains/RestaurantCategoryFilter.jsx";
+import RestaurantList from "./components/mains/RestaurantList.jsx";
+import RestaurantInfoModal from "./components/asides/RestaurantInfoModal.jsx";
+import AddRestaurantModal from "./components/asides/AddRestaurantModal.jsx";
+import restaurants from "./data/restaurants.js";
 
 function App() {
-  return <h1>Self-Paced React</h1>;
+  const [selectedCategory, setSelectedCategory] = useState("전체");
+
+  const filteredRestaurants =
+    selectedCategory === "전체"
+      ? restaurants
+      : restaurants.filter((restaurant) => restaurant.category === selectedCategory);
+
+  return (
+    <>
+      <Gnb />
+      <RestaurantCategoryFilter
+        selectedCategory={selectedCategory}
+        onCategoryChange={setSelectedCategory}
```

Comment:
저는 setter함수(`setSelectedCategory`이고 setter,getter할 때의 그것과 다릅니다!) 를 prop으로 넘길 때는 setter함수 이름 그대로 넘기고 받는 것을 선호합니다. 그래야만 이것이 setter함수인지 일반함수인지 확실하게 읽기 쉬워서요!
```suggestion
       setSelectedCategory={setSelectedCategory}
```

Reply:
양쪽에 같은 변수명을 쓰는 것 같은 느낌(?) 때문에 최대한 비슷한 의미를 갖는 변수를 사용했는데 { }로 감싸져 있기 때문에 같은 변수로 인식하지 않는군요!

Reply:
왼쪽은 단지 속성명이며 오른쪽은 실제 함수에요. 스터디 시간에 나눴던 이야기로 이해가 되셨을거라고 생각합니다!